### DISCUSSION TITLE
Group 2 packages in one MR

### DISCUSSION THREAD

**Discussion by 123Haynes** [renovatebot/renovate#12941] [Request Help] [answered]
Hello,

I'm currently using renovate 29.8.5 and I'm trying to group 2 packages together so only 1 MR will be created by renovate.

My `renovate.json` looks like this.
```json
{
    "extends": [
        "config:base",
        ":semanticCommits",
        ":semanticCommitTypeAll(fix)"
    ],
    "vulnerabilityAlerts": {
        "labels": ["security"]
    },
    "packageRules": [
        {
            "groupName": "Spring Boot and Spring Cloud",
            "matchPackagePrefixes": ["org.springframework.cloud:", "org.springframework.boot:"]
        }
    ]
}
```

But with this the 2 MRs for Spring Cloud and Spring Boot are just being closed automatically.

The packages in question are:
`org.springframework.boot:spring-boot-starter`
`org.springframework.cloud:spring-cloud-dependencies`

What do I need to change so both dependencies are grouped into one MR?

**Comment by rarkins** [comment, marked as answer]
It's possible that you're doing everything right, but the new (group) PR is simply blocked due to prHourlyRateLimit=2 and you already created 2 PRs this hour

**Reply by 123Haynes** [reply]
@rarkins Thank you :)    
That was indeed one of the problems.  
The other one was that one dependency had a major update and the other just a minor update so renovate generated different branches for them.  

I changed my config like this and now it groups both dependencies into one MR 😄 

```json
{
    "extends": [
        "config:base",
        ":semanticCommits",
        ":semanticCommitTypeAll(fix)",
        ":prHourlyLimitNone"
    ],
    "vulnerabilityAlerts": {
        "labels": ["security"]
    },
    "packageRules": [
        {
            "groupName": "Spring Boot and Spring Cloud",
            "matchPackagePrefixes": [
                "org.springframework.cloud:",
                "org.springframework.boot:"
            ],
            "separateMajorMinor": false
        }
    ]
}
```

**Comment by friesoft** [comment]
I'm trying exactly the same, using the same config (from solution) and for me I'm getting multiple pullrequests every time :(

First pullrequest (output parts from debug log):
```
                 "depName": "org.springframework.boot:spring-boot-maven-plugin",
                 "currentValue": "2.4.4",
                     "newValue": "2.6.1",

                 "depName": "org.springframework.boot:spring-boot-starter-parent",
                 "currentValue": "2.4.4",
                     "newValue": "2.6.1",
```

Second pullrequest: (output parts from debug log)
```
                 "depName": "org.springframework.cloud:spring-cloud-dependencies",
                 "currentValue": "2020.0.3",
                     "newValue": "2021.0.0",
```

Using this config
```
module.exports = {
  "extends": [
    "config:base",
    ":prHourlyLimitNone"
  ],
  "packageRules": [
    {
      "groupName": "Spring",
      "matchPackagePrefixes": ["org.springframework.cloud:", "org.springframework.boot:"],
    }
  ]
};
```

Using renovate 31.2.2 with Bitbucket Server

Any ideas what I'm doing wrong? I've tried matchPackagePatterns (org.springframework.*), disabling separateMajorMinor, separateMinorPatch and separateMultipleMajor

Nothing seems to do the trick :( Any help appreciated

**Reply by rarkins** [reply]
What are the two branch names you see?

**Reply by friesoft** [reply]
renovate/spring-cloud
renovate/spring-boot

**Reply by rarkins** [reply]
If you add `ignorePresets: ["group:recommended"]` to your config then do you get a different result?

**Reply by friesoft** [reply]
No difference :( Also tried adding the relevant spring groups to ignorePresets
```
module.exports = {
  "extends": [
    "config:base",
    ":prHourlyLimitNone"
  ],
  "ignorePresets": ["group:recommended", "group:springBoot", "group:springCloud"],
  "packageRules": [
    {
      "groupName": "Spring",
      "matchPackagePrefixes": ["org.springframework.cloud:", "org.springframework.boot:"],
      "separateMajorMinor": false,
      "separateMinorPatch": false,
      "separateMultipleMajor": false
    }
  ]
};
```

**Reply by rarkins** [reply]
The problem is you are specifying presets in your global config instead of repo config. Put them in renovate.json instead

**Reply by friesoft** [reply]
Aren't global configurations like this possible? We only have java applications so it would be duplicate effort to put them in every repository.

**Reply by viceice** [reply]
You should use a shared preset for multiple repos.

**Reply by friesoft** [reply]
Thank you all for your help. To be honest I don't really get yet why some config only works in the repo renovate.json vs the renovate-config.js but I'm only some hours into renovate so maybe I missed this part in the docs. Thanks!

**Reply by TorstenF76** [reply]
strip the colon from `org.springframework.boot`

the name of the plugin ends after `boot`, there is no colon in your gradle file.
